src/librvnpal/src/posix/mapping.c

```diff
@@ -48,17 +49,32 @@ rvn_create_and_mmap64_file(
 {
     int32_t rc = SUCCESS;
 
+    assert(path);
     assert(initial_file_size > 0);
 
-    _ensure_path_exists(path);
-      
-    struct map_file_handle* mfh = calloc(1, sizeof(struct map_file_handle));
+    struct map_file_handle *mfh = calloc(1, sizeof(struct map_file_handle));
     *handle = mfh;
     if(mfh == NULL)
     {
         rc = FAIL_CALLOC;
         goto error_cleanup;
     }
+
+    if(path[0] == '\0')
```

Comment:
How can that be? What would be passing us something like this?
And if they are passing us an empty string, wouldn't this cause issues with `strdup` never ending?

Reply:
https://github.com/ravendb/ravendb/pull/8970#discussion_r266767782
@avivadi commented that we should accept an empty string and it should reference to current directory as `"."`. As I understand it `strdup` has no problem with an empty string but `stat` do
so I switch it with `"."`.

Reply:
Consider the case where we send an empty C# string `""`.
This is translated by pinvoke to `"\0"` as a C string.
You are writing changing this to be: `"."`, but you _don't have a null terminator.

Now you pass a string without null terminator to `strdup` and will cause _issues_.

You can _change_ the value, so `if (path[0] == '\0') path = "."; `, which _is_ null terminated.

Reply:
Also add a comment explaining this explicitly